### PR TITLE
Improve error for when instance lock is already locked

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

Improves the error message for when the instance lock is already locked by adding the lock path on the message.

## Why is it important?

It makes understanding the error easier as it'll state exactly what is locked.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Have 2 filebeats running using the same `path.data`, the second should show the error that this PR improve 

## Related issues

- Relates elastic/elastic-agent#1772

## Logs

Original log:
```
[elastic_agent][error] filebeat stderr: "Exiting: data path already locked by another beat. Please make sure that multiple beats are not sharing the same data path (path.data).\n"
```

### CODE REVIEW

File: libbeat/cmd/instance/locks/lock.go [hunk 1]
```diff
@@ -111,8 +111,9 @@ func (lock *Locker) Lock() error {
 	}
 	// case: lock could not be obtained.
 	if !isLocked {
+
```
Comment (belimawr) [inline libbeat/cmd/instance/locks/lock.go:114]: Nitpick
```suggestion
```

**Review by belimawr — APPROVED** [review]

File: changelog/fragments/1669212543-improve-error-for-when-instance-lock-is-already-locked.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: enhancement
+
+# Change summary; a 80ish characters long description of the change.
+summary: Adds the file path of the instance lock on the error when it's is already locked
```
Comment (cmacknz) [inline changelog/fragments/1669212543-improve-error-for-when-instance-lock-is-already-locked.yaml:14]: Beats doesn't use the changelog tool yet so this won't do anything.

You would need to add an entry in CHANGELOG.next.asciidoc at the root of the repo.

We could also just omit a changelog entry for a change this small.

**Review by cmacknz — APPROVED** [review]

**Pull request merged**